### ISSUE TITLE
GPU issue (100%)

### ISSUE THREAD

**Issue by VolyFnap** [holoiso-eol/holoiso#94]
GPU залочен на 300mhz (37-46w), есть способ разблочить? (RX 580)

**Comment by MichaelMKenny** [comment]
I'm also having my RX 560m performance capped when playing in the deck UI. Hope there's a fix. It only started happening earlier today 🤔

**Comment by MichaelMKenny** [comment]
I seemed to have worked around the issue by editing config.vdf in `~/.steam/steam/config/config.vdf` while in desktop mode with Steam quit. I changed "TDPLimitEnabled" to "1", and "TDPLimit" to "35", which seems to give me full performance again in deck UI.

**Comment by 420247paul** [comment]
MichaelMKenny i cant find these entries my file looks like its full of jiberish

**Comment by MichaelMKenny** [comment]
> MichaelMKenny i cant find these entries my file looks like its full of jiberish

It's right at the top of the file with all the other deck specific settings. You might have to change one of the those settings in the deck UI first for them to appear in the config file. To change one of those settings, you need to change into the deck UI, launch a game, press the guide button + A button on your controller, go to the battery tab, go to advanced view and change some settings, probably the TDP limit checkbox (and slider). Hopefully then when you go back out the desktop, the config.vdf will have those entries at the top.

**Comment by 420247paul** [comment]
MichaelMKenney i couldnt get it to show up i enabled a TDP limit still nothing, was wondering if you dont mind copy and pasting your file so i could compare it with mine

**Comment by MichaelMKenny** [comment]
> MichaelMKenney i couldnt get it to show up i enabled a TDP limit still nothing, was wondering if you dont mind copy and pasting your file so i could compare it with mine

I'd rather not post the entire file, but here is the first 33 lines (with the settings in question):

```
"InstallConfigStore"
{
	"Software"
	{
		"Valve"
		{
			"Steam"
			{
				"AutoUpdateWindowEnabled"		"0"
				"Perf"
				{
					"GPUPerformanceLevel"		"1"
					"GraphicsProfilingEnabled"		"0"
					"OverlayEnabled"		"0"
					"OverlayLevel"		"1"
					"FrameRateLimitEnabledv2"		"0"
					"FPSLimit"		"60"
					"VariableResolutionEnabled"		"0"
					"DynamicRefreshRateEnabled"		"0"
					"TDPLimitEnabled"		"1"
					"CPUGovernor"		"1"
					"CPUGovernorManualMhz"		"1500"
					"ScalingFilter"		"1"
					"FSRSharpness"		"1"
					"ShowPerfOverlayOverSteam"		"0"
					"LowLatencyEnabled"		"0"
					"FPSLimitv2"		"60"
					"TDPLimit"		"30"
				}
				"ipv6check_http_state"		"good"
				"ipv6check_udp_state"		"bad"
				"ShaderCacheManager"
				{
```

**Comment by rbrugnolli** [comment]
What is happening is that there was an update that in the config.vdf file this type of configuration no longer appears.
It changed completely, I got to do what was indicated, but after I restarted the pc it is already completely changed.

**Comment by 420247paul** [comment]
you have to close steam befor you edit the file rbrugnolli keep trying it works

**Comment by 420247paul** [comment]
nevermind im lying mine reverted too

**Comment by rbrugnolli** [comment]
So, now if you edit the config.vdf a lot when you open steam it redoes the file, it's probably now saved in some other way or another file.

**Comment by massatt212** [comment]
Does this fix really work, don't want to reinstall and be disappointed again.

**Comment by 420247paul** [comment]
yes it works dont put 30 put your cards tdp and dont change the slider or it reverts and you have to do it again

**Comment by rbrugnolli** [comment]
Yesterday an update came out, if the update doesn't work anymore.

https://www.theverge.com/2022/5/11/23067870/steam-deck-update-per-game-performance-profiles

**Comment by massatt212** [comment]
anyone know where the new files are going for separate game settings?

**Comment by MichaelMKenny** [comment]
I've done a bit of initial digging in changing the TDP in the config file, since the update that added per-game settings. Attached is a document with my findings so far:

[SteamOS 3 Custom TDP (After Per-Game Profile Update).pdf](https://github.com/theVakhovskeIsTaken/holoiso/files/8684912/SteamOS.3.Custom.TDP.After.Per-Game.Profile.Update.pdf)

**Comment by rbrugnolli** [comment]
Interesting, you can configure using the profiles, I even managed to make it work but you have to remove the profiles, leave the way it is just above.
But the way the document would be correct.

**Comment by massatt212** [comment]
whats hex code for 120w?

**Comment by rbrugnolli** [comment]
78

**Comment by massatt212** [comment]
ill go test it, if it doesnt work ill delete everything inside their and lock the file.

**Comment by massatt212** [comment]
i dont understand this

"InstallConfigStore"
{
	"Software"
	{
		"Valve"
		{
			"Steam"
			{
				"perf"
				{
					"GameProfiles"
					{
						"769"
						{
							"0"
							{
								"0"		"0d0000803f1001180120012802300138c801400048005000580f600168dc0b70017805800100880100900100980100a0013c"
							}
						}
					}
				}
				"AutoUpdateWindowEnabled"
				{
				}
				"ipv6check_http_state"		"good"
				"ipv6check_udp_state"		"good"
				"ShaderCacheManager"
				{

**Comment by rbrugnolli** [comment]
"0d0000803f1001180120012802300138c80140004800500058**78**600168dc0b70017805800100880100900100980100a0013c"

Put it this way

**Comment by massatt212** [comment]
didnt work, i have a rx470 anyone got tips?

**Comment by rbrugnolli** [comment]
GameProfiles"
{
"769"
{
"0"
{
"0" "0d0000803f1001180120012802300138c801400048005000580f600168dc0b70017805800100880100900100980100a0013c"
}
}
}

In mine I did it like this, I removed all that part and put it the way MichaelMKenny put the log above, here it worked, but the profile separated by game is lost.
In TDPLimit put the 120w.

**Comment by massatt212** [comment]
im sure valve will release theirs end of the year, or until some one figures this out properly ill wait.

**Comment by massatt212** [comment]
i have seen people making addon plugins for steam deck, can some one make a plugin to edit tdp from GPU and CPU

**Comment by sysdba1977** [comment]
i have sucsessfully get tdp 120w on my rx 6600 xt. i replaced all above perfomance in config file to shader cache manager line with old code in top of page. in replaced code set tdp 120 not in hex. go to steam session and see 120 w in tdp) maybe this will help someone

**Comment by massatt212** [comment]
Send me your file

**Comment by sysdba1977** [comment]
it for rx 6600 xt. here it is
[config.vdf.zip](https://github.com/theVakhovskeIsTaken/holoiso/files/8713178/config.vdf.zip)
it for tdp 120 w and fps 143. you can edit for another tdp and another fps limit

**Comment by massatt212** [comment]
ok ill test it out tmr

**Comment by sysdba1977** [comment]
i set my tdp to 160 w on rx 6600 xt. it boost in games to 2735 mhz. all work perfect)

**Comment by massatt212** [comment]
got my gpu to work properly, but games work worst than normal manjaro arch

**Comment by massatt212** [comment]
is it possible to change mesa driver to regular mesa?
when i tried i get this jupiter crash error

**Comment by pastaq** [comment]
This issue is caused by the file ```/usr/bin/steamos-polkit-helpers/steamos-priv-write``` enabling access to a bunch of stuff that should be root only in most situations. It is part of the ```jupiter-hw-support``` package. Removing that package with ```yay -R jupiter-hw-support``` and a reboot will probably solve all these problems and should, IMO, be removed by default as the deckui has a lot of hard coded automations that mess with these files, as you can see.

**Comment by FlamingScar** [comment]
TLDR: i fixed it by installing CoreCtrl, an app normally used for overclocking, but in this case I just used it to force my GPU to run at stock clock speeds

I fixed this issue in a much different way

for me, none of the above fixes worked

I was having the issue in both desktop and gaming mode

for desktop mode installing CoreCtrl and setting my clock speeds to the max was enough

for gaming mode I had to follow the instructions here: https://gitlab.com/corectrl/corectrl/-/wikis/Setup#dont-ask-for-user-password so CoreCtrl didnt ask for the root password, which from what I can tell doesn't work inside of gaming mode (in desktop mode right click Corectrl in the application launcher in the bottom left and click "add to steam" to be able to access it in gaming mode)

youll have to launch corectrl every time you swap from gaming mode to desktop mode or vise versa (unless your only going to play games in gaming mode which in that case you'll only need to ever launch it in gaming mode) and if you restart your system

after a bit of playing I noticed that leaving corectrl running in the background was causing crashing, luckily closing it doesn't reset the core clocks back to 500mhz, also a few times corecrtl would reset the core clocks back to 500mhz when I first open gaming mode, so be sure to check that corectrl is putting the clock speed to the max, or at least your gpus stock speeds

**Comment by massatt212** [comment]
i got my tdp to hit 100w on a rx 470 and im still loosing fps, i notice gamescope makes me loose up to 20fps in some games

**Comment by jj1425** [comment]
Hi guys, after an afternoon of trial and error with all the suggested solutions to this problem I thought I'd share what worked for me - I wasn't happy with how limiting CoreCtrl was in this set up so I installed [https://github.com/azeam/powerupp](url) and set it up with the default profile, and saved the settings as a persistent save, which creates a bash script and a udev rules file that loads the script on system startup. This seems to be working for me across both gamescope and desktop mode as long as I don't touch the built in TDP settings. Fingers crossed it sticks!!

**Comment by pastaq** [comment]
See this comment for a good way to prevent this as well: https://github.com/theVakhovskeIsTaken/holoiso/issues/294#issuecomment-1167051433
